include/accessor.h

```diff
@@ -469,17 +532,104 @@ class accessor<DataT, Dims, Mode, target::host_task> : public detail::accessor_b
+		if(detail::is_prepass_handler(cgh)) {
+			auto& prepass_cgh = dynamic_cast<detail::prepass_handler&>(cgh);
+			auto rmfn = access::all();
+			prepass_cgh.add_requirement(detail::get_buffer_id(buff), std::make_unique<detail::range_mapper<0, decltype(rmfn)>>(rmfn, Mode, buff.get_range()));
+		} else {
+			if(detail::get_handler_execution_target(cgh) != detail::execution_target::host) {
+				throw std::runtime_error(
+				    "Calling accessor constructor with host_buffer target is only allowed in host tasks."
+				    "If you want to access this buffer from within a parallel_for task, please specialize the call using one of the non host tags");
+			}
+			auto access_info =
+			    detail::runtime::get_instance().get_buffer_manager().access_host_buffer<DataT, 0>(detail::get_buffer_id(buff), Mode, subrange<0>());
+			m_host_ptr = static_cast<DataT*>(access_info.ptr);
+		}
+	}
+
+	template <access_mode TagModeNoInit>
+	accessor(const buffer<DataT, 0>& buff, handler& cgh, const detail::access_tag<Mode, TagModeNoInit, target::host_task> /* tag */) : accessor(buff, cgh) {}
+
+	/**
+	 * TODO: As of ComputeCpp 2.5.0 they do not support no_init prop, hence this constructor is needed along with discard deduction guide.
+	 *    but once they do this should be replace for a constructor that takes a prop list as an argument.
+	 */
+	template <access_mode TagMode, access_mode M = Mode, typename = std::enable_if_t<detail::access::mode_traits::is_producer(M)>>
+	accessor(
+	    const buffer<DataT, 0>& buff, handler& cgh, const detail::access_tag<TagMode, Mode, target::host_task> /* tag */, const property::no_init /* no_init */)
+	    : accessor(buff, cgh) {}
+
+	template <access_mode TagMode, access_mode TagModeNoInit>
+	accessor(const buffer<DataT, 0>& buff, handler& cgh, const detail::access_tag<TagMode, TagModeNoInit, target::host_task> /* tag */,
+	    const property_list& /* prop_list */) {
+		static_assert(detail::constexpr_false<std::integral_constant<access_mode, TagMode>>,
+		    "Currently it is not accepted to pass a property list to an accessor constructor. Please use the property celerity::no_init "
+		    "as a last argument in the constructor");
+	}
+
+	template <access_mode M = Mode>
+	std::enable_if_t<detail::access::mode_traits::is_producer(M), DataT&> operator*() const {
+		return *m_host_ptr;
+	}
+
+	template <access_mode M = Mode>
+	std::enable_if_t<detail::access::mode_traits::is_pure_consumer(M), const DataT&> operator*() const {
+		return *m_host_ptr;
+	}
+
+	friend bool operator==(const accessor& lhs, const accessor& rhs) { return lhs.m_host_ptr == rhs.m_host_ptr; }
+
+	friend bool operator!=(const accessor& lhs, const accessor& rhs) { return !(lhs == rhs); }
+
+  private:
+	DataT* m_host_ptr = nullptr;
+};
+
+
+template <typename T, int D, access_mode Mode, access_mode ModeNoInit, target Target>
+accessor(const buffer<T, D>& buff, handler& cgh, const detail::access_tag<Mode, ModeNoInit, Target> tag) -> accessor<T, D, Mode, Target>;
```

Comment:
I feel like it would be clearer if these wouldn't be templated on `D` at all (since `D = 0` anyway), and came after the guides for `D > 0`, in the same order the classes are implemented.

Reply:
Done.